### ISSUE TITLE
FMB001 new firmware issues in packet decoding

### ISSUE THREAD

**Issue by durduliu2009** [traccar/traccar#4216]
Hello guys,

I have a Teltonika FMB001 tracker device with a new firmware version.
In this new version in avl packet there are added more values from device. As far as I know in this new firmware they (Teltonika) implemented codec8_extended (https://wiki.teltonika.lt/view/Codec_8_Extended) 

In this mooment on new firmware there are "some options" to enable extra parameters to send (like ON/OFF) that have codec 8 extended implemented 

If those extra parameters that have codec 8 extended are disabled everything is fine ... and when those extra parameters are enabled i see some errors in traccar server log 

```
2019-02-05 08:21:04  INFO: [936e1c97: 5027 < 109.166.137.125] HEX: 000f333532303934303833363739373736
2019-02-05 08:21:04  INFO: [936e1c97: 5027 > 109.166.137.125] HEX: 01
2019-02-05 08:21:04  INFO: [936e1c97: 5027 < 109.166.137.125] HEX: 00000000000003558e0600000168be600b81020f89f9a71a8e1497006d01610e000000f70001000100f705000000000000000000000168be600799020f89f9a71a8e1497006d01610e000000f70001000100f705000000000000000000000168be6003b1020f89f9a71a8e1497006d01610e000000f70001000100f705000000000000000000000168be5fffc9020f89f9a71a8e1497006d01610e000000f70001000100f705000000000000000000000168be5ffbe1020f89f9a71a8e1497006d01610e000000f70001000100f705000000000000000000000168be5ff411020f89f9a71a8e1497006d01610f000000f70002000100f705000000000000000101010258ffad001e001bffa6001a001bffb1001e001cffa8001c0017ffa90019001dffac001a0017ffa6001c0017ffac00170018ffa800140019ffa900180017ffad001d0012ffa500140018ffae001b001affa8001e0015ffaa001c0019ffab001b001dffaa001c0018ffa900190017ffa8001c0018ffa900190017ffa8001a0018ffa9001d0016ffab001d001bffa500190017ffaa001d0015ffa7001a0016ffaa0019001bffa7001a0015ffa8001c0016ffa800180019ffa8001a0019ffab001d0012ffa900180015ffaa0015001bffa800180016ffa900190017ffaa00190018ffaa001a001dffab001b0019ffaa00170017ffaa00160018ffa900190018ffac00190018ffa800180016ffaa0018001affa800180019ffac001c0018ffa900190015ffab00190017ffa8001a0018ffab001c001cffa9001a0015ffa8001a0019ffab00200018ffad001b0017ffa7001d0015ffab001a001affa7001a0014ffab001a0011ffa6001a0015ffaa001a0016ffa9001b0016ffab00180018ffa900160019ffa900150017ffa900180016ffa90017001bffa900150016ffa900170018ffa9001a0014ffa800150016ffa700160019ffaa001a0015ffaa001b0019ffab00180019ffa9001b0016ffa800190015ffa8001a001affa7001a0014ffa9001d0014ffa8001e0019ffa8001a0017ffa8001c0018ffa9001b0016ffa8001e0017ffab0016001affa8001c0018ffa8001b0016ffa90017001affa9001a0018ffa800190014ffa800180018ffa900190018ffa900170018ffaa00180016ffaa001a0015ffac00180018ffa80017001affab00180017ffac001700160600004830
2019-02-05 08:21:04  WARN: [936e1c97] error - readerIndex(863) + length(4) exceeds writerIndex(865): PooledSlicedByteBuf(ridx: 863, widx: 865, cap: 865/865, unwrapped: PooledUnsafeDirectByteBuf(ridx: 865, widx: 865, cap: 1024)) - IndexOutOfBoundsException (... < TeltonikaProtocolDecoder:94 < *:205 < *:257 < *:391 < *:443 < *:486 < *:476 < ExtendedObjectDecoder:51 < ... < WrapperContext:102 < ...)
```
Basically when extra parameters are enabled on device this error appear when those parameters are disabled everything is fine
Can someone help ? 
10x

**Comment by tananaev** [comment]
What version of Traccar are you using?

**Comment by durduliu2009** [comment]
4.1

> What version of Traccar are you using?

4.1

**Comment by tananaev** [comment]
You have to test on latest one before raising issues.
